Modules/Loadable/Markups/MRML/vtkSlicerDijkstraGraphGeodesicPath.cxx

```diff
@@ -0,0 +1,188 @@
+/*==============================================================================
+
+  Copyright (c) Laboratory for Percutaneous Surgery (PerkLab)
+  Queen's University, Kingston, ON, Canada. All Rights Reserved.
+
+  See COPYRIGHT.txt
+  or http://www.slicer.org/copyright/copyright.txt for details.
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  This file was originally developed by Kyle Sunderland, PerkLab, Queen's University
+  and was supported through CANARIE's Research Software Program, Cancer
+  Care Ontario, OpenAnatomy, and Brigham and Women�s Hospital through NIH grant R01MH112748.
+
+==============================================================================*/
+
+// Markups MRML includes
+#include "vtkSlicerDijkstraGraphGeodesicPath.h"
+
+// VTK includes
+#include <vtkFloatArray.h>
+#include <vtkPointData.h>
+#include <vtkInformation.h>
+#include <vtkInformationVector.h>
+
+//------------------------------------------------------------------------------
+vtkStandardNewMacro(vtkSlicerDijkstraGraphGeodesicPath);
+
+//------------------------------------------------------------------------------
+vtkSlicerDijkstraGraphGeodesicPath::vtkSlicerDijkstraGraphGeodesicPath()
+{
+  this->UseScalarWeights = true;
+  this->RecalculateAdjacency = true;
+  this->CostFunction = COST_FUNCTION_DISTANCE;
+}
+
+//------------------------------------------------------------------------------
+vtkSlicerDijkstraGraphGeodesicPath::~vtkSlicerDijkstraGraphGeodesicPath()
+= default;
+
+//------------------------------------------------------------------------------
+void vtkSlicerDijkstraGraphGeodesicPath::PrintSelf(std::ostream &os, vtkIndent indent)
+{
+  Superclass::PrintSelf(os, indent);
+  os << indent << "CostFunction: " << this->GetCostFunctionAsString(this->CostFunction) << std::endl;
+}
+
+//------------------------------------------------------------------------------
+const char* vtkSlicerDijkstraGraphGeodesicPath::GetCostFunctionAsString(int costFunction)
+{
+  switch (costFunction)
+    {
+    case COST_FUNCTION_DISTANCE:
+      {
+      return "distance";
+      }
+    case COST_FUNCTION_ADDITIVE:
+      {
+      return "additive";
+      }
+    case COST_FUNCTION_INVERSE_SQUARED:
+      {
+      return "inverseSquared";
+      }
+    default:
+      {
+      return "";
+      }
+    }
+}
+
+//------------------------------------------------------------------------------
+const char* vtkSlicerDijkstraGraphGeodesicPath::GetCostFunctionAsHumanReadableString(int costFunction)
```

Comment:
It should be GetCostFunctionFromString and do just enum<->code string conversion. Human-readable string <-> enum conversion should happen in the GUI layer (same as for the other similar function).